File: src/main/java/org/codegas/stores/domain/value/NodeType.java
package org.codegas.stores.domain.value;

public enum NodeType {
    ENTRANCE,
    CHECKOUT,
    REGULAR;

    public static NodeType defaultNonSingleton() {
        return REGULAR;
    }

    public boolean isSingleton() {
        return this != REGULAR;
    }
}

File: codegas-grapp/src/main/webapp/app/storelayout/model/NodeType.js
(function() {
   "use strict";

   angular.module("App")
      .constant("NodeType", createEnum());

   function createEnum() {
      return {
         ENTRANCE: new NodeType("E", "Entrance", "content/img/marker_green.png", true),
         CHECKOUT: new NodeType("C", "Checkout", "content/img/marker_red.png", true),
         REGULAR: new NodeType("R", "Regular", "content/img/marker_blue.png", false)
      };
   }

   function NodeType(code, displayString, iconUrl, isSingleton) {
      this.code = code;
      this.displayString = displayString;
      this.iconUrl = iconUrl;
      this.isSingleton = isSingleton;
   }
})();